### PR TITLE
:sparkles: agent no longer hardcoded to java

### PR DESCRIPTION
**Pull request by djzager**
Attempt at some simple logic to not hardcode our programming language as Java without rewriting everything.

## Summary by CodeRabbit

* **New Features**
  * Automatic programming-language detection from file paths/URIs to provide context-aware assistance.
  * Dynamic, language-specific dependency and import guidance instead of a generic default.

* **Tests**
  * Comprehensive unit tests validating language detection and LLM-friendly language mapping across common file types and edge cases.

### CODE REVIEW

File: shared/src/utils/languageMapping.ts [hunk 1]
```diff
@@ -373,3 +373,120 @@ export function getLanguageDisplayName(language: string): string {
...
+
+  // Special case: Java build/config files should map to Java
+  if (
+    lowerFilename === "pom.xml" ||
+    lowerFilename === "build.gradle" ||
+    lowerFilename === "build.gradle.kts" ||
+    lowerFilename.endsWith(".gradle")
+  ) {
+    return "Java";
+  }
```
[the remark this answers is not in the document]
Comment (djzager) [reply shared/src/utils/languageMapping.ts]: I updated the parsing and added tests for windows paths
